src/core/execute.c

```diff
@@ -4659,6 +4703,15 @@ static int exec_child(
                         return log_unit_error_errno(unit, r, "Failed to apply system call filters: %m");
                 }
 #endif
+
+#if HAVE_LIBBPF
+                r = apply_restrict_filesystems(unit, context);
```

Comment:
hmm, the other bpf hooks are applied via the cgroups logic, i.e. see how CGROUP_CONTROLLER_BPF_SOCKET_BIND works. I think that's a better approach, since we want to install the bpf program when realizing the cgroup and not when forking off a process, since we might fork off many processes, and we shouldn't update the cgroup each time.

Reply:
The BPF program is installed only once when creating the system manager. It's true that with current code we're updating the global BPF map with every process we fork off so that's not ideal.

If we move the code to the cgroups logic we'd need to allow some filesystems so things are correctly set up like tmpfs for STDIN or cgroup to attach the cgroup. And if we allow them temporarily and removed them later we're doing a map operation anyway so we might as well not restrict until the last moment.

Reply:
oh, of course, that makes sense. (might be cool to add a comment to the definition of CGroupController enum that quickly mentions this before the next person looking at this code gets confused)